project-koku/koku
# Source not found loop

Sources can get stuck in a loop during message processing if a source is not found in sources-api backend. This PR skips message processing if the source isn't found.

Testing:
1. apply diff:
```diff
diff --git a/koku/sources/kafka_listener.py b/koku/sources/kafka_listener.py
index 3c11ca39..e4db3ec8 100644
--- a/koku/sources/kafka_listener.py
+++ b/koku/sources/kafka_listener.py
@@ -366,6 +366,7 @@ async def process_message(app_type_id, msg, loop=EVENT_LOOP):  # noqa: C901
         None
 
     """
+    await asyncio.sleep(10)
     LOG.info(f"Processing Event: {str(msg)}")
     msg_data = None
     try:
```
2. complete Source create and delete POSTs during sleep. See message skipped:
```
sources_client    | [2020-04-15 02:40:23,456] INFO None Cost Management Message to process: {'event_type': 'Application.create', 'offset': 92, 'source_id': 1, 'auth_header': '...'}
sources_client    | [2020-04-15 02:40:33,457] INFO None Processing Event: {'event_type': 'Application.create', 'offset': 92, 'source_id': 1, 'auth_header': '...'}
sources_client    | [2020-04-15 02:40:33,476] WARNING None Source not found in platform sources: {'event_type': 'Application.create', 'offset': 92, 'source_id': 1, 'auth_header': '...'}
```
3. see message processing continue
